### ISSUE TITLE
SharpDx error on win10 

### ISSUE THREAD

**Issue by xlp100** [helix-toolkit/helix-toolkit#330]
i want to run sharpdx example,but always error
the code is 
```
       private void StartD3D()
        {
            if (activeClients != 0)
                return;

            context = new Direct3DEx();

            var presentparams = new PresentParameters
            {
                Windowed = true,
                SwapEffect = SwapEffect.Discard,
                DeviceWindowHandle = GetDesktopWindow(),
                PresentationInterval = PresentInterval.Default,    
            };
           
  device = new DeviceEx(context, this.adapterIndex, DeviceType.Hardware, IntPtr.Zero, CreateFlags.HardwareVertexProcessing | CreateFlags.Multithreaded | CreateFlags.FpuPreserve, presentparams);
        }
```
the error message is
SharpDX.SharpDXException: HRESULT: [0x8876086C], Module: [SharpDX.Direct3D9], ApiCode: [D3DERR_INVALIDCALL/InvalidCall], Message: Unknown
   In SharpDX.Result.CheckError()
   IN SharpDX.Direct3D9.Direct3DEx.CreateDeviceEx(Int32 adapter, DeviceType deviceType, IntPtr hFocusWindow, Int32 behaviorFlags, PresentParameters[] presentationParametersRef, DisplayModeEx[] fullscreenDisplayModeRef, DeviceEx returnedDeviceInterfaceOut)

i was setup DXSDK_Jun10,now how can i fixed it.

**Comment by kthyholdt** [comment]
I have this issue as well on one computer. With the latest package on MyGet. The package on Nuget has not had this problem.
I have found that it is the GetBestAdapter() method that is failing to get the best adapter. It is returning the Windows Basic Renderer instead of my Nvidia adapter.

The reason it is chosen over the Nvidia adapter is that videoMemory for my Nvidia GTX 970 is returned as -1073741824 while for the Windows driver it is returned as 0.

long videoMemory = item.Description.DedicatedVideoMemory;

The same behavior has been discussed here: http://www.gamedev.net/topic/679144-sharpdx-dxgi-adapter-video-memory-problem/

**Comment by kthyholdt** [comment]
I have looked further into this, and it crashes on the following line in StartD3D() in DX11ImageSource.cs

`device = new DeviceEx(context, this.adapterIndex, DeviceType.Hardware, IntPtr.Zero, CreateFlags.HardwareVertexProcessing | CreateFlags.Multithreaded | CreateFlags.FpuPreserve, presentparams);`

It crashes when this.adapterIndex is pointing to the WARP driver.

If exchanged with this line it works.

`device = new DeviceEx(context, 0, DeviceType.Hardware, IntPtr.Zero, CreateFlags.HardwareVertexProcessing | CreateFlags.Multithreaded | CreateFlags.FpuPreserve, presentparams);`

It was changed in commit 61b0be4a.
I have found that it also works if run as x64 instead of x86. Then the videoMemory will be returned as positive. And the Nvidia will be the best Adapter.
